activesupport/test/option_merger_test.rb

```diff
@@ -84,13 +84,20 @@ def test_nested_method_with_options_containing_hashes_going_deep
     end
   end
 
-  def test_nested_method_with_options_using_lambda
+  def test_nested_method_with_options_using_lambda_as_only_argument
     local_lambda = lambda { { lambda: true } }
     with_options(@options) do |o|
       assert_equal @options.merge(local_lambda.call), o.method_with_options(local_lambda).call
     end
   end
 
+  def test_proc_as_first_argument_with_other_options_should_still_merge_options
+    p = proc { }
+    with_options(@options) do |o|
+      assert_equal [[p], @options.merge({ a: "b" })], o.method_returning_args_and_options(p, { a: "b" })
+    end
```

Comment:
Single-letter names make this slightly harder to digest.  What about mirroring the other tests using the following?

```suggestion
    local_proc = proc { }
    local_options = { "cool" => true }
    
    with_options(@options) do |o|
      assert_equal [local_proc, @options.merge(local_options)], o.method_with_args(local_proc, local_options)
    end
```